crates/walrus-sui/src/client/retry_client/retriable_rpc_client.rs

```diff
@@ -603,3 +639,13 @@ impl From<tonic::Status> for CheckpointRpcError {
         }
     }
 }
+
+impl CheckpointRpcError {
+    pub(crate) fn is_checkpoint_not_produced(&self) -> bool {
+        self.status.code() == tonic::Code::NotFound
+            && self
+                .status
+                .checkpoint_height()
+                .is_some_and(|height| self.checkpoint_seq_num.unwrap_or(0) > height)
```

Comment:
Should this be `>=`? Does `self.checkpoint_seq_num` represent the last loaded checkpoint, or the next checkpoint? If it's already loaded `self.checkpoint_seq_num`, then we should change `>` to `>=`, I think?

Reply:
`self.checkpoint_seq_num` is the checkpoint that the walrus node tries to read.